### ISSUE TITLE
Tipref not altering lazer position on certain tooltypes

### ISSUE THREAD

**Issue by Redd56** [Yellow-Dog-Man/Resonite-Issues#2656]
### Describe the bug?

On certain tools (lighttool and meshtool are the 2 that comes to mind) dont respect the tipref position when it comes to lazer position.

### To Reproduce

make a custom light tool or mesh tool with a tipref that isnt in the default spot
see how the lazer doesnt move

### Expected behavior

lazer to follow tipref position like all the other tools (dev tool, custom tool, flux tool etc) that way the light tool and mesh tool (and any others that have this issue maybe microphone might) can have custom lazer positions

### Screenshots

![image](https://github.com/user-attachments/assets/bb077990-dac6-424a-9006-4277fbbd45b7)
![image](https://github.com/user-attachments/assets/7c886f40-0f87-4706-8cf1-cb36f2ac1942)

### Resonite Version Number

Beta 2024.7.22.1319

### What Platforms does this occur on?

Windows, Linux

### Log Files

[DESKTOP-5DC208J - 2024.7.22.1319 - 2024-07-23 22_14_05.log](https://github.com/user-attachments/files/16357541/DESKTOP-5DC208J.-.2024.7.22.1319.-.2024-07-23.22_14_05.log)

### Additional Context

i think i remember someone once saying its a feature apparently which seems kinda silly. as if someone makes a tooltip that isnt just a small little doodad that sits at the tip of your finger its prolly gonna have a custom position for the tip meaning moving the lazer should happen

### Reporters

redd56

**Comment by Redd56** [comment]
tips that do respect tooltipref position: dev, material, flux, rawdata, meter, color, label,glue,grababble setter, character colider, mesh vissibility, interactive cam exclude, mesh tool, sun mover,duplicator, 
ones that dont: light, shape tool, microphone tool, geometry line brush, controller diagnostic(doesnt even do anything for that tool),object slicer,rig mesh transfer tool, all brush tools,particle spray, wigler aplyer

**Comment by Redd56** [comment]
also fluxtips wirepoint is located at tipref as well instead of wirepoint

**Comment by shiftyscales** [comment]
Thank you.

**Comment by Redd56** [comment]
oh wirepoint is for the fluxwire not the cutting wire

**Comment by Redd56** [comment]
Should i just split this into a bunch of individal issues so they can be worked on incrimentaly?

**Comment by Redd56** [comment]
Also i noticed some of the tools in resonite essentials like the light tool have custom tip refeneces. So this would also fix those issues if they arent made already

**Comment by Redd56** [comment]
Currently the: 
* all brush tools
* controller diagnostic
* geometry line brush
* light tool
* microphone tool
* object slicer
* particle spray
* rig mesh transfer tool
* Rig tool
* shape tool
* spotlight tool
* wiggler applier

all dont modify the laser position

@ProbablePrime

**Comment by Redd56** [comment]
also i labeled it wrong as it seems some of the tools actually do use it.
but the issue is about it not modifying the users laser position.

**Comment by ProbablePrime** [comment]
Ill take a look at what I can do here, it might be problematic to upgrade items or existing tools though, will leave it assigned to me.

**Comment by Redd56** [comment]
yeah some tools might need to have a new field added for their visual like the brush tools since those use the tooltiprefrence as the place where the brush exists

**Comment by Redd56** [comment]
Add the new rig tool to the list
